config/auth/iap/service_account_auth_helper.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2019, Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package config
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2/jws"
+
+	oauth "golang.org/x/oauth2/google"
+)
+
+const (
+	audURL                  = "https://www.googleapis.com/oauth2/v4/token"
+	jwtBearerTokenGrantType = "urn:ietf:params:oauth:grant-type:jwt-bearer"
+	expirationSecs          = 3600
+)
+
+// GetIDTokenWithServiceAccount gets an IAP ID Token required for authenticating a service account with IAP.
+// For more info, see https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_service_account
+func GetIDTokenWithServiceAccount(config IapConfig) (string, error) {
+	creds, err := ioutil.ReadFile(config.ServiceAccountKeyPath)
+	if err != nil {
+		return "", fmt.Errorf("could not read service account creds file: %v", err)
+	}
+
+	// Creating and signing JWT token.
+
+	jwtc, err := oauth.JWTConfigFromJSON(creds)
+	if err != nil {
+		return "", fmt.Errorf("could not parse config from service account creds: %v", err)
+	}
+
+	pk, err := parseKey(jwtc.PrivateKey)
+	if err != nil {
+		return "", fmt.Errorf("invalid private json key: %v", err)
+	}
+
+	now := time.Now().Unix()
+
+	claims := &jws.ClaimSet{
+		Aud: audURL,
+		Iss: jwtc.Email,
+		Sub: jwtc.Email,
+		Iat: now,
+		Exp: now + expirationSecs,
+		PrivateClaims: map[string]interface{}{
+			"target_audience": config.IapClientId,
+		},
+	}
+
+	h := &jws.Header{
+		Algorithm: "RS256",
+		Typ:       "JWT",
```

Comment:
there's a defaultHeader in jwt with these two constants included.. no defined constants though. Probably better to leave it like this instead of copy the default header